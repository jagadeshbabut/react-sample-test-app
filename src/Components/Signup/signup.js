import React, {Component, Fragment} from 'react';
import {Link} from 'react-router-dom';

class Signup extends Component {
    constructor(props) {
        super(props);
        this.state = {}
    }

    onChange = (event) => {
        this.setState({
            [event.target.name]: event.target.value
        });
    }

    signup = (event) => {
        event.preventDefault();
        let user_info = this.state
        delete user_info['repassword']
        sessionStorage.setItem('user_info', JSON.stringify(user_info));
        this
            .props
            .history
            .push('/signin');
    }

    render() {
        return (
            <Fragment>
                <form className="form-signup" onSubmit={(e) => this.signup(e)}>
                    <img
                        className="mb-4"
                        src="//getbootstrap.com/docs/4.3/assets/brand/bootstrap-solid.svg"
                        alt=""
                        width="72"
                        height="72"/>
                    <h1 className="h3 mb-3 font-weight-normal">Please sign up</h1>
                    <label htmlFor="name" className="sr-only">Name</label>
                    <input
                        type="text"
                        id="name"
                        name="name"
                        className="form-control"
                        placeholder="Name"
                        onChange={(e) => this.onChange(e)}
                        required
                        autoFocus/>
                    <label htmlFor="inputEmail" className="sr-only">Email address</label>
                    <input
                        type="email"
                        id="inputEmail"
                        name="email"
                        className="form-control"
                        placeholder="Email address"
                        onChange={(e) => this.onChange(e)}
                        required/>
                    <label htmlFor="inputPassword" className="sr-only">Choose Password</label>
                    <input
                        type="password"
                        id="inputPassword"
                        name="password"
                        className="form-control"
                        placeholder="Choose a Password"
                        onChange={(e) => this.onChange(e)}
                        required/>
                    <label htmlFor="reInputPassword" className="sr-only">Re-confirm Password</label>
                    <input
                        type="password"
                        id="reInputPassword"
                        name="repassword"
                        className="form-control"
                        placeholder="Re-confirm Password"
                        onChange={(e) => this.onChange(e)}
                        required/>
                    <button className="btn btn-lg btn-primary btn-block" type="submit">Sign Up</button>
                </form>
                <hr/>
                <p>
                    <Link to="/signin">Login if already signed up!</Link>
                </p>
            </Fragment>
        )
    }
}

export default Signup;