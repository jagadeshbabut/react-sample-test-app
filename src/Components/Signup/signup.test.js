import React from 'react';
import {shallow, mount} from 'enzyme';
import Signup from './signup';
import {MemoryRouter} from 'react-router-dom';

describe('Signup Component', () => {
    it('Renders without crashing', () => {
        shallow(<Signup/>)
    });

    it('Signup component has not changed', () => {
        const wrapper = shallow(<Signup/>);
        expect(wrapper).toMatchSnapshot()
    });

    it('Check Signin link is working', () => {
        const REDIRECTED = '/signin'
        const wrapper = mount(
            <MemoryRouter initialEntries={[REDIRECTED]}>
                <Signup/>
            </MemoryRouter>
        );
        expect(wrapper.find('Link')).toBeTruthy();
        wrapper
            .find('Link')
            .simulate('click');
        expect(wrapper.find('Link').prop('to')).toEqual(REDIRECTED)
    });

});