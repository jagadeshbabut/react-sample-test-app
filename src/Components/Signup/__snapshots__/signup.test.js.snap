// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Component Signup component has not changed 1`] = `
<Fragment>
  <form
    className="form-signup"
    onSubmit={[Function]}
  >
    <img
      alt=""
      className="mb-4"
      height="72"
      src="//getbootstrap.com/docs/4.3/assets/brand/bootstrap-solid.svg"
      width="72"
    />
    <h1
      className="h3 mb-3 font-weight-normal"
    >
      Please sign up
    </h1>
    <label
      className="sr-only"
      htmlFor="name"
    >
      Name
    </label>
    <input
      autoFocus={true}
      className="form-control"
      id="name"
      name="name"
      onChange={[Function]}
      placeholder="Name"
      required={true}
      type="text"
    />
    <label
      className="sr-only"
      htmlFor="inputEmail"
    >
      Email address
    </label>
    <input
      className="form-control"
      id="inputEmail"
      name="email"
      onChange={[Function]}
      placeholder="Email address"
      required={true}
      type="email"
    />
    <label
      className="sr-only"
      htmlFor="inputPassword"
    >
      Choose Password
    </label>
    <input
      className="form-control"
      id="inputPassword"
      name="password"
      onChange={[Function]}
      placeholder="Choose a Password"
      required={true}
      type="password"
    />
    <label
      className="sr-only"
      htmlFor="reInputPassword"
    >
      Re-confirm Password
    </label>
    <input
      className="form-control"
      id="reInputPassword"
      name="repassword"
      onChange={[Function]}
      placeholder="Re-confirm Password"
      required={true}
      type="password"
    />
    <button
      className="btn btn-lg btn-primary btn-block"
      type="submit"
    >
      Sign Up
    </button>
  </form>
  <hr />
  <p>
    <Link
      replace={false}
      to="/signin"
    >
      Login if already signed up!
    </Link>
  </p>
</Fragment>
`;
